lib/matplotlib/axes/_axes.py

```diff
@@ -5327,12 +5315,27 @@ def imshow(self, X, cmap=None, norm=None, aspect=None,
             The Colormap instance or registered colormap name used to map
             scalar data to colors. This parameter is ignored for RGB(A) data.
 
-        norm : `~matplotlib.colors.Normalize`, optional
+        norm : str or `~matplotlib.colors.Normalize`, optional
             The `.Normalize` instance used to scale scalar data to the [0, 1]
             range before mapping to colors using *cmap*. By default, a linear
             scaling mapping the lowest value to 0 and the highest to 1 is used.
             This parameter is ignored for RGB(A) data.
 
+            The normalization can also be given as a str, which should be a
+            scale name (as in `~.Axes.set_xscale`, i.e. one of "linear", "log",
+            "symlog", "logit", etc.). In that case, a normalization class is
+            dynamically generated from the corresponding scale, as if using
+            that scale for the image's colorbar.
```

Comment:
```suggestion
            If a str, *norm* must be a scale name (i.e. "linear", "log",
            "symlog", "logit", as per `~.Axes.set_xscale`, or call 
            `~.scale.get_scale_names`). By default, this scale will be 
            also used for a colorbar for the image.  
```

Reply:
I'm not really sure about this wording?  (Why "by default"?  1) I guess you *could* change that, but having a linear-scale colorbar for a lognorm seems rare; 2) more than anything, this kwarg sets the norm on the mappable; setting the colorbar scale is more a side effect of that.  Also, even set_xscale doesn't mention `get_scale_names`.  Perhaps it should, but if it doesn't, I don't think imshow() needs to do that either.)

Reply:
"one of "linear", "log", "symlog", "logit", etc." is vague, so was trying to be more precise.  "...as if using the scale" was also theoretical sounding, whereas we _do_ use the scale for the colorbar, so why not be explicit?  As for "by default" - the user can change the colorbar scale.    But I don't feel strongly about my wording - feel free to iterate as you see fit.